src/library/scala/collection/BufferedIterator.scala

```diff
@@ -24,5 +24,10 @@ trait BufferedIterator[+A] extends Iterator[A] {
    */
   def head: A
 
+  def headOption : Option[A] = {
+    if (this.nonEmpty) Option(head)
+    else None
+  }
+
```

Comment:
IMHO, it looks better as a one-liner using `hasNext`. And `Some(head)`.

https://github.com/scala/scala/blob/v2.12.0-M5/src/library/scala/collection/TraversableLike.scala#L412